### PR TITLE
Refactor/code quality

### PR DESCRIPTION
**Pull request by danieleades**
general code quality refactoring

not sure if you prefer these commits separate or squashed?

### CODE REVIEW

File: src/plantuml_backend_factory.rs [hunk 1]
```diff
@@ -11,30 +11,27 @@ use reqwest::Url;
...
     } else {
-        Box::new(PlantUMLShell::new(cmd.clone()))
+        Box::new(PlantUMLShell::new(cmd.to_string()))
     }
 }
 
 #[cfg(not(any(feature = "plantuml-ssl-server", feature = "plantuml-server")))]
-fn create_backend(cmd: &String) -> Box<dyn PlantUMLBackend> {
+fn create_backend(cmd: &str) -> Box<dyn PlantUMLBackend> {
     Box::new(PlantUMLShell::new(cmd.clone()))
```
Comment (sytsereitsma) [inline src/plantuml_backend_factory.rs:36]: Compilation fails here:
```
   Compiling mdbook-plantuml v0.7.0 (C:\Projects\mdbook-plantuml)
error[E0308]: mismatched types
  --> src\plantuml_backend_factory.rs:36:33
   |
36 |     Box::new(PlantUMLShell::new(cmd.clone()))
   |                                 ^^^^^^^^^^^
   |                                 |
   |                                 expected struct `std::string::String`, found `&str`
   |                                 help: try using a conversion method: `cmd.to_string()`
```

**Review by sytsereitsma — COMMENTED** [review]
Nice work, I'm learning a few things too.
Couple of things I do not like, see my remarks below.

File: tests/e2e_test/e2e_tester.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def change_output_dir(book_toml, output_dir):
 
 
 def _build_book(book_name, open_browser=True, output_dir=None, clean=True):
```
[the remark this answers is not in the document]
Comment (danieleades) [reply tests/e2e_test/e2e_tester.py:22]: true, i did not run the e2e tests.

i think the simplest thing is for me to simply revert that commit

File: src/plantuml_backend_factory.rs (hunk 1 above)
Comment (danieleades) [reply src/plantuml_backend_factory.rs]: what flags are you using to see that compilation error? and what platform are you using?

it should be fixed now

File: src/dir_cleaner.rs [hunk 3]
```diff
@@ -32,39 +32,37 @@ pub struct DirCleaner {
...
 
-    pub fn keep(&mut self, img_path: &PathBuf) {
-        info!("DirCleaner - Keeping {}", img_path.to_string_lossy());
+    pub fn keep(&mut self, img_path: &Path) {
+        log::info!("DirCleaner - Keeping {}", img_path.to_string_lossy());
         self.files.remove(img_path);
     }
 
-    fn get_files(img_path: &PathBuf) -> HashSet<PathBuf> {
+    fn files(img_path: &Path) -> HashSet<PathBuf> {
```
[the remark this answers is not in the document]
Comment (danieleades) [reply src/dir_cleaner.rs]: i've dropped that commit for now

File: src/plantuml_backend_factory.rs (hunk 1 above)
Comment (sytsereitsma) [reply src/plantuml_backend_factory.rs]: No special flags. Just ```cargo build``` and ```cargo build --release```
Using rust 1.59.0

**Pull request merged**